packages/next/src/server/base-server.ts

```diff
@@ -889,13 +889,15 @@ export default abstract class Server<ServerOptions extends Options = Options> {
         )
       }
 
-      req.headers['x-forwarded-host'] ??= req.headers['host'] ?? this.hostname
-      req.headers['x-forwarded-port'] ??= this.port?.toString()
       const { originalRequest } = req as NodeNextRequest
-      req.headers['x-forwarded-proto'] ??= (originalRequest.socket as TLSSocket)
-        ?.encrypted
-        ? 'https'
-        : 'http'
+      const isHttps = !!(originalRequest.socket as TLSSocket)?.encrypted
```

Comment:
```suggestion
      const isHttps = !!(originalRequest?.socket as TLSSocket)?.encrypted
```

This should fix the broken tests.